client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/RollupDocumentationIT.java

```diff
@@ -160,4 +173,164 @@ public void onFailure(Exception e) {
             assertTrue(latch.await(30L, TimeUnit.SECONDS));
         }
     }
+
+    public void testGetRollupCaps() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+
+        DateHistogramGroupConfig dateHistogram =
+            new DateHistogramGroupConfig("timestamp", DateHistogramInterval.HOUR, new DateHistogramInterval("7d"), "UTC"); // <1>
+        TermsGroupConfig terms = new TermsGroupConfig("hostname", "datacenter");
+        HistogramGroupConfig histogram = new HistogramGroupConfig(5L, "load", "net_in", "net_out");
+        GroupConfig groups = new GroupConfig(dateHistogram, histogram, terms);
+        List<MetricConfig> metrics = new ArrayList<>(); // <1>
+        metrics.add(new MetricConfig("temperature", Arrays.asList("min", "max", "sum")));
+        metrics.add(new MetricConfig("voltage", Arrays.asList("avg", "value_count")));
+
+        //tag::x-pack-rollup-get-rollup-caps-setup
+        final String indexPattern = "docs";
+        final String rollupIndexName = "rollup";
+        final String cron = "*/1 * * * * ?";
+        final int pageSize = 100;
+        final TimeValue timeout = null;
+
+        String id = "job_1";
+        RollupJobConfig config = new RollupJobConfig(id, indexPattern, rollupIndexName, cron,
+            pageSize, groups, metrics, timeout);
+
+        PutRollupJobRequest request = new PutRollupJobRequest(config);
+        PutRollupJobResponse response = client.rollup().putRollupJob(request, RequestOptions.DEFAULT);
+
+        boolean acknowledged = response.isAcknowledged();
+        //end::x-pack-rollup-get-rollup-caps-setup
+        assertTrue(acknowledged);
+
+        // Now that the job is created, we should have a rollup index with metadata.
+        // We can test out the caps API now.
```

Comment:
Does the test rely on the existence of the rollup index? If so we need to wait for its creation

Reply:
I'll add a check just in case, but it shouldn't strictly be necessary.  The PutJob API won't return until the index has been created, and the CreateIndex API doesn't return successfully until the CS is updated and the primaries are allocated.

The caps info is stored in the metadata of an index, so as long as the PutJob API returns successfully it should be there.